rest/data/aliases.go

```diff
@@ -14,9 +14,20 @@ type DBAliasConnector struct{}
 
 // FindProjectAliases queries the database to find all aliases.
 // If the repoId is given, we default to repo aliases if there are no project aliases.
-func (d *DBAliasConnector) FindProjectAliases(projectId, repoId string) ([]restModel.APIProjectAlias, error) {
+// If aliasesToAdd are given, then we fold those aliases in and remove any that are marked as deleted.
+func (d *DBAliasConnector) FindProjectAliases(projectId, repoId string, aliasesToAdd []restModel.APIProjectAlias) ([]restModel.APIProjectAlias, error) {
```

Comment:
This function seems really complex now. When would the caller need to optionally add or remove aliases? Can that happen separately?

Reply:
Yeah I was torn about how to handle this... This is to handle the routes for modifying project/repo, so the aliases that are being added/removed in the route are cased on out here, rather than individually in each route, especially since we get aliases back as APIProjectAliases. I decided this was the best option, because for places where we don't care about this we can just pass in nil aliasesToAdd. The alternative I'll go with if we really don't like this is to create a helper function in one of the rest/route files, and that could be used in both API modify routes, but I prefer having it all here.